lib/index.ts

```diff
@@ -85,10 +85,38 @@ class PWMetrics {
         const sheets = new Sheets(this.sheets, this.clientSecret);
         await sheets.appendResults(results.runs);
       }
+
+      if (this.resultHasExpectationErrors(results) && this.flags.expectations) {
+        throw new Error(messages.getMessage('HAS_EXPECTATION_ERRORS'));
+      }
     }
     return results;
   }
 
+  resultHasExpectationErrors(results: PWMetricsResults): boolean {
+    const expectationErrors = results.runs.filter(run => {
+      return run.timings.filter(item => {
```

Comment:
What if instead of looping through every result for a missed expectation you do it immediately after each run?
https://github.com/willmendesneto/pwmetrics/blob/c41b753f6dbdcac8da15916151a0907134cfabd0/lib/index.ts#L70

Something like
```js
try {
 const lhMetric: MetricsResults = await this.run();
 lhMetric.timings
   .filter(metric => metric.timing >= expectations[metric.id])
   .forEach(metric => {
     // Here you should have access to each failed metric for a specific run.
     // You can set a flag to false as the end state (if already false dont touch) 
     // like @paulirish suggested, there is probably a more performant method than looping
     // through every timing in every run.
   });
  // check if any failed, filter(...).length > 0 ? and return accordingly?
 metricsResults[runIndex] = lhMetric;
 console.log(messages.getMessageWithPrefix('SUCCESS', 'SUCCESS_RUN', runIndex, runs.length));
}
```

I do have a question, if you ask pwmetrics to execute n runs, should you fail on the first not meet expectation, should you test the expectations only for the median run or something in between?